tests/test_visitors/test_ast/test_annotations/test_union_annotations.py

```diff
@@ -0,0 +1,37 @@
+import pytest
+
+from wemake_python_styleguide.violations.best_practices import (
+    DisallowUnionTypeViolation,
+)
+from wemake_python_styleguide.visitors.ast.annotations import (
+    WrongAnnotationVisitor,
+)
+
+functions_breaking_rule = [
+    """def function(a: Union[int, str]): ...""",
+    """def function(a: typing.Union[int, str]): ...""",
+    """def function(a: t.Union[int, str]): ...""",
+    """def function(a: List[Union[int, str]]): ...""",
+    """def function(a: int) -> Union[int, str]: ...""",
+    """def function(a: Optional[int]) -> None: ...""",
+]
+
+
+@pytest.mark.parametrize('function', functions_breaking_rule)
```

Comment:
You should test `python3.9-` here explicitly.

Reply:
Done by checking the Python version and expecting no errors if the version is <3.10